subprocess/subtree_oom.go

```diff
@@ -0,0 +1,180 @@
+package subprocess
+
+import (
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type OOMTracker struct {
+	WasOOMKilled bool  `json:"was_oom_killed"`
+	IsSudo       bool  `json:"is_sudo"`
+	Pids         []int `json:"pids"`
+}
+
+func NewOOMTracker() *OOMTracker {
+	oomTracker := OOMTracker{}
+	return &oomTracker
+}
+
+func (o *OOMTracker) Clear() error {
+	return o.clear()
+}
+
+func (o *OOMTracker) Check() error {
+	return o.check()
+}
+
+func (o *OOMTracker) clear() error {
+	var err error
+	switch os := runtime.GOOS; os {
+	case "linux":
+		err = o.clearLinux()
+		break
+	case "darwin":
+		err = o.clearDarwin()
+		break
+	default:
+		return errors.New("os does not handle OOM")
+	}
+
+	return errors.Wrap(err, "error in clear")
+}
+
+func (o *OOMTracker) setIsSudo() error {
+	err := exec.Command("sudo", "date").Run()
+	if err != nil {
+		switch err.(type) {
+		case *exec.ExitError:
+			o.IsSudo = false
```

Comment:
`o.IsSudo` is already false, so need to set. Also no need to `break`. You can just `// pass` or return nil.